docs/storage-surface.rst

```diff
@@ -0,0 +1,916 @@
+
+In the case that the bucket has existing blobs, the backend
+will return a `409 conflict`_ and raise
+
+.. code-block:: python
+
+   >>> storage.delete_bucket(bucket_name)
+   Traceback (most recent call last):
+     File "<stdin>", line 1, in <module>
+   gcloud.exceptions.Conflict: 409 Some Message
+
+this can be addressed by using the ``force`` keyword
+
+   >>> storage.delete_bucket(bucket_name, force=True)
+
+Even using ``force=True`` will fail if the bucket contains more than 256
+blobs. In this case, delete the blobs manually before deleting the bucket.
+
+.. _409 conflict: http://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_Client_Error
+
+Working with Buckets
+--------------------
+
+To create a bucket object directly
+
+.. code-block:: python
+
+   >>> bucket = storage.Bucket('bucket-name')
+   >>> bucket.exists()
+   False
+   >>> bucket.create()
+   >>> bucket.exists()
+   True
+
+You can also use an explicit connection when calling
+:class:`Bucket <gcloud.storage.bucket.Bucket>` methods which make HTTP
+requests
+
+.. code-block:: python
+
+   >>> bucket = storage.Bucket('bucket-name')
+   >>> bucket.exists(connection=connection)
+   False
+   >>> bucket.create(connection=connection)
+
+By default, just constructing a :class:`Bucket <gcloud.storage.bucket.Bucket>`
+does not load any of the associated bucket metadata. To load all bucket
+properties
+
+.. code-block:: python
+
+   >>> bucket = storage.Bucket('bucket-name')
+   >>> print bucket.last_sync
+   None
+   >>> print bucket.self_link
+   None
+   >>> bucket.reload()
+   >>> # May be necessary to include projection and fields for last sync
+   >>> bucket.last_sync
+   datetime.datetime(2015, 1, 1, 12, 0)
+   >>> bucket.self_link
+   u'https://www.googleapis.com/storage/v1/b/bucket-name'
+
+Instead of calling
+:meth:`Bucket.reload() <gcloud.storage.bucket.Bucket.reload>`, you
+can load the properties when the object is instantiated by using the
+``eager`` keyword
+
+.. code-block:: python
+
+   >>> bucket = storage.Bucket('bucket-name', eager=True)
+   >>> bucket.last_sync
+   datetime.datetime(2015, 1, 1, 12, 0)
+
+As with using :class:`Bucket <gcloud.storage.bucket.Bucket>` methods,
+you can also pass an explicit ``connection`` in when using ``eager=True``.
+
+To delete a bucket
+
+.. code-block:: python
+
+   >>> bucket.delete()
+
+or
+
+.. code-block:: python
+
+   >>> bucket.delete(force=True)
+
+as above.
+
+To make updates to the bucket use
+:meth:`Bucket.patch() <gcloud.storage.bucket.Bucket.patch>`
+
+.. code-block:: python
+
+   >>> bucket.versioning_enabled = True
+   >>> bucket.patch()
+
+If there are no updates to send, an exception will occur
+
+.. code-block:: python
+
+   >>> bucket.patch()
+   Traceback (most recent call last):
+     File "<stdin>", line 1, in <module>
+   ValueError: No updates to send.
+
+In total, the properties that can be updated are
+
+.. code-block:: python
+
+   >>> bucket.acl = [
```

Comment:
Is there a better way to handle ACLs? This seems to require a lot of strings (like `'OWNER'` and `'user-...'`). Is the `.grant_....` stuff no good?